helix-term/src/commands.rs

```diff
@@ -2337,10 +2337,12 @@ mod cmd {
         args: &[&str],
         _event: PromptEvent,
     ) -> anyhow::Result<()> {
+        use helix_core::path::expand_tilde;
         let id = view!(cx.editor).doc;
 
         if let Some(path) = args.get(0) {
-            cx.editor.open(path.into(), Action::VerticalSplit)?;
+            cx.editor
+                .open(expand_tilde(Path::new(path)), Action::VerticalSplit)?;
```

Comment:
This is the wrong place to fix, instead we want to expand tilde at the start of canonicalize (instead of at the end when we normalize https://github.com/helix-editor/helix/blob/b99db7c6875f74a15dcfb0cfdb3334c837d4aab1/helix-core/src/path.rs#L43)

I'll write a quick patch